packages/organization/src/organization.ts

```diff
@@ -0,0 +1,148 @@
+        fireAndForget.add(() => revokeIntegration(integration.externalId, integration.type));
+      });
+    }
+  };
+  const removeUserFromOrganization = async (
+    orgId: string,
+    allUsers: OrganizationWithUsers[`users`],
+    tier: Tier,
+    subOrganizations: OrganizationWithUsers[],
+  ): Promise<void> => {
+    const maxUsersInTier = tierConstraints[tier].maxUsers;
+    if (tier !== Tier.Launch && tier !== Tier.Build && tier !== Tier.Grow) {
+      return;
+    }
+
+    const sortedUsers = Array.from(new Set(allUsers)).sort(
+      (a, b) => new Date(a.user.createdAt).getTime() - new Date(b.user.createdAt).getTime(),
+    );
+
+    const usersToKeep = sortedUsers.slice(0, maxUsersInTier);
+
+    const adminInKeptUsers = usersToKeep.some((user) => user.role === OrganizationRoleEnum.ORG_ADMIN);
+
+    if (!adminInKeptUsers) {
+      const oldestAdmin = sortedUsers.find((user) => user.role === OrganizationRoleEnum.ORG_ADMIN);
+      if (oldestAdmin) {
+        usersToKeep.pop();
+        usersToKeep.push(oldestAdmin);
+      }
+    }
+
+    const usersToRemove = sortedUsers.filter((user) => !usersToKeep.some((keep) => keep.userId === user.userId));
+    const subOrgIds = subOrganizations.map((subOrg) => subOrg.id);
+    const organizationIds = [orgId, ...subOrgIds];
+
+    await prisma.userOrganization.deleteMany({
+      where: {
+        userId: { in: usersToRemove.map((userOrg) => userOrg.userId) },
+        organizationId: { in: organizationIds },
+      },
+    });
+
+    for (const subOrg of subOrganizations) {
+      const subOrgAdmin = Array.from(new Set(subOrg.users))
+        .sort((a, b) => new Date(a.user.createdAt).getTime() - new Date(b.user.createdAt).getTime())
+        .find((user) => user.role === OrganizationRoleEnum.ORG_ADMIN);
+      if (!subOrgAdmin) {
+        const rootOrgAdmin = usersToKeep
+          .sort((a, b) => new Date(a.user.createdAt).getTime() - new Date(b.user.createdAt).getTime())
+          .find((user) => user.role === OrganizationRoleEnum.ORG_ADMIN);
+
+        if (rootOrgAdmin) {
+          await prisma.userOrganization.update({
+            where: { userId_organizationId: { userId: rootOrgAdmin.userId, organizationId: subOrg.id } },
+            data: { role: OrganizationRoleEnum.ORG_ADMIN },
+          });
+        }
+      }
+    }
+  };
+
+  const organizations = await prisma.organization.findMany({
+    where: { OR: [{ id: organizationId }, { parentId: organizationId }] },
+    ...organizationWithUsers,
+  });
+
+  const currentTier = organizations[0].tier;
+
+  if (currentTier === newTier) {
+    return organizations[0];
+  }
+
+  const subOrganizations = organizations.filter((org) => org.parentId === organizationId);
+
+  const allUsers = organizations.flatMap((orgItem) => orgItem.users);
+  const maxIntegrations = tierConstraints[newTier].maxIntegrations;
+
+  await removeUserFromOrganization(organizationId, allUsers, newTier, subOrganizations);
```

Comment:
```suggestion
  await removeUserFromOrganization(allUsers, newTier, organizations);
```
Why are you passing subOrganization here? Shouldn't you remove the users from the root organization as well?